supabase/functions/alert-data-processing/index.ts

```diff
@@ -0,0 +1,218 @@
+import { serve } from "https://deno.land/std@0.184.0/http/server.ts";
+import { isEmpty, isFinite } from "https://cdn.skypack.dev/lodash";
+
+import { corsHeaders } from "../_shared/cors.ts";
+import { returnPostgresError } from "../_shared/helpers.ts";
+import { supabaseClient } from "../_shared/supabaseClient.ts";
+
+interface DataProcessingArguments {
+    bytes_processed: number;
+    emails: string[];
+    evaluation_interval: string;
+    spec_type: string;
+}
+
+interface AlertRecord {
+    alert_type: string;
+    catalog_name: string;
+    fired_at: string;
+    resolved_at: string | null;
+    arguments: DataProcessingArguments;
+}
+
+interface EmailConfig {
+    emails: string[];
+    subject: string;
+    content: string;
+}
+
+export const handleSuccess = <T>(response: any) => {
+    return response.error
+        ? {
+            data: null,
+            error: response.error,
+        }
+        : {
+            data: response.data as T,
+        };
+};
+
+export const handleFailure = (error: any) => {
+    return {
+        data: null,
+        error,
+    };
+};
+
+const dataProcessingAlertType = "data_not_processed_in_interval";
+
+const TABLES = { ALERT_HISTORY: "alert_history" };
+
+const getTaskDetailsPageURL = (catalogName: string, specType: string) =>
+    `https://dashboard.estuary.dev/${specType}s/details/overview?catalogName=${catalogName}`;
+
+const emailNotifications = async (
+    pendingNotifications: EmailConfig[],
+    token: string,
+    senderAddress: string,
+): Promise<void> => {
+    const notificationPromises = pendingNotifications.map(
+        ({ emails, content, subject }) =>
+            fetch("https://api.resend.com/emails", {
+                method: "POST",
+                headers: {
+                    ...corsHeaders,
+                    "Content-Type": "application/json",
+                    "Authorization": `Bearer ${token}`,
+                },
+                body: JSON.stringify({
+                    from: senderAddress,
+                    to: emails,
+                    subject,
+                    html: `
+                      <div style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;">
+                        ${content}
+
+                        <p style="margin-bottom: 0;">Thanks,</p>
+                        <p style="margin-top: 0;">Estuary Team</p>
+                      </div>
+                    `,
+                }),
+            }),
+    );
+
+    await Promise.all(notificationPromises);
+};
+
+serve(async (_request: Request): Promise<Response> => {
+    const startTimestamp = new Date();
+    const minuteOffset = startTimestamp.getUTCMinutes() - 5;
```

Comment:
One other thing I just noticed: This can result in `minuteOffset` being negative, and the behavior of `setUTCMinutes` seems pretty surprising in that case.

Reply:
This should not be a problem given the only use of `minuteOffset` is when it is passed to `setUTCMinutes()`, which handles positive and negative integers as expected (e.g., `date.setUTCMinutes(-3)` results in the minute value of the `date` being `57`).

Reply:
It should not impact the correctness of the edge function since `minuteOffset` will only be negative when the current time is within five minutes of the hour.

But, to your point, I do believe many people would be surprised by the behavior of `setUTCMinutes`.